app/bundles/ReportBundle/Model/ReportExporter.php

```diff
@@ -81,6 +81,29 @@ public function processExport(ExportOption $exportOption)
     private function processReport(Scheduler $scheduler)
     {
         $report = $scheduler->getReport();
+
+        if (!is_null($scheduler->getScheduleDate())) {
```

Comment:
I think that this logic should be moved to Scheduler class itself:
- I would rename getter to something like `if ($scheduler->adjustScheduleDate()))`
- Clone of DateTime class should be performed inside of Scheduler class.
- Scheduler class should return correct DateTime directly (move switch to the Scheduler class).

Reply:
Hello @Maxell92 

`$scheduler->getScheduleDate()` should be set all the time at this point, this condition is here "just in case" to fall back to the original process.

I don't think this should be moved to Schedueler class, these dates modifications aren't linked to scheduler, but to the report that is currently exported, it looks like the right place to put them into reportExportOptions class

What do your think about this ?

Reply:
Hello @Dcoutelle,

well you are right that suggested solution is not ideal too - it should be a new class with responsibility of adjusting dates, otherwise there is a problem with SRP.

But I still do not think this code should be part of reportExporter. Can you add tests for this? It will not be strait-forward, because you need to test lot of conditions here.
But it will be easy to write tests for a couple of methods in Scheduler entity - actually a few data providers for testing correct date should be enough.

Reply:
I do not see any advantage of doing this - method will be called inside the class, so it has to be private. Private methods cannot be tested. And the issue with SRP still remains.